synapse/handlers/federation.py

```diff
@@ -294,6 +294,16 @@ def backfill(self, dest, room_id, limit, extremities=[]):
             extremities=extremities,
         )
 
+        # Don't bother processing events we already have.
```

Comment:
Could you explain why this might happen and what the callers will do in this case?